packages/headless/src/api/search/search-api-client.test.ts

```diff
@@ -320,13 +332,6 @@ describe('search api client', () => {
         logger,
         origin: 'searchApiFetch',
         requestParams: {
-          analytics: {
-            capture: state.configuration.analytics.analyticsMode === 'next',
-            clientId: expect.any(String),
-            clientTimestamp: expect.any(String),
-            documentReferrer: state.configuration.analytics.originLevel3,
-            originContext: state.configuration.analytics.originContext,
-          },
```

Comment:
Indeed, the rest of the changes are 👌 and rightly point to the fact that the coverage was lacking, or tests being false positive in some instance.

This one maybe point to an issue in the code if you had to remove the analytics part to make the test pass ?

Reply:
Ah, I had actually added this particular chunk myself in https://github.com/coveo/ui-kit/pull/3770.

I had done this because initially, I was replacing `expect(request).toMatchObject(expectedRequest);` with `expect(request).toEqual(expectedRequest);` to make the test more reliable, which meant that the `expectedRequest` object had to have the exact same properties as the actual request. I ultimately took a different approach: I made `PlatformClientCallOptions` generic so that we can impose stricter typing on the `requestParams` property (which was previously typed as `unknown`) while still testing for only subsets of non-required request params.

In this PR, I decided to remove the `analytics` request param from this test for now because it's not a primitive and it can't simply be copied over from the state into the `expectedRequest` object like `context`, for instance. Also, the property would not be passed in the request at all if analytics is not enabled in the search engine configuration.

I think we should have specific search API client tests for the `analytics` request parameter.

There are several request parameters (not only in this call) that I think we should test more thoroughly in this suite.

This highlights a problem in the test coverage of our various buildXYZRequest functions. These functions are essential because they bridge the headless state and the Coveo API. Yet, from what I can see, several of those functions only have cursory tests in search-api-client.test, which means it would be pretty easy to introduce breaking changes or bugs wihtout any tests failing, especially since the spread syntax we use in those functions makes everything very error prone...

Reply:
>  but it would definitely be better to have better coverage in Headless too 😅

Ideally, project should not even compile in such a scenario, that's the best protection, and there's no reason we cannot achieve this I think.

Second best scenario is UT in Headless would catch it.

3rd best is e2e, as you pointed out.

Reply:
I created [KIT-3082](https://coveord.atlassian.net/browse/KIT-3082) and [KIT-3083](https://coveord.atlassian.net/browse/KIT-3083) to deal with compilation and test coverage respectively